File: FRONTEND/app/src/main/java/com/myscratch/app/domain/model/Note.kt
package com.myscratch.app.domain.model

data class Note(
    val id: String,
    val userId: String,
    val folderId: String,
    val title: String,
    val content: String,
    val isLocked: Boolean = false,
    val isDeleted: Boolean = false,
    val deletedAt: Long? = null,
    val updatedAt: Long = System.currentTimeMillis(),
    val createdAt: Long = System.currentTimeMillis()
)

File: BACKEND/app/Models/Note.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Note extends Model
{
    protected $fillable = [
        'user_id',
        'folder_id',
        'title',
        'content',
        'is_pinned',
    ];

    protected $casts = [
        'title' => 'encrypted',
        'content' => 'encrypted',
        'is_pinned' => 'boolean',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function folder()
    {
        return $this->belongsTo(Folder::class);
    }
}
